### PR TITLE
test(stack): use jsdom for unit testing

### PR DESCRIPTION
**Pull request by vvo**
Basically I switched to jsdom because:
- mocha cannot show nice diffs when assertions throws (https://gi
hub.com/mochajs/mocha/issues/1348)
- karma + mocha can be really slow
- you should only need a real browser in unit testing when debugging
weird stuff, so you have now npm run `test:watch:browser`
- integration testing will handle the "multi browser test"

I also added expect.toEqualJSX to get nice diffs.

Some before/after:

before:
> npm test (11.5s)
2015-10-16-102143_784x375_scrot

after:
> npm test (4.5s)
2015-10-16-102109_765x398_scrot

before:
> JSX failure in npm test
2015-10-16-102450_1037x566_scrot

after:
> JSX failure in npm test
2015-10-16-102602_760x613_scrot

### CODE REVIEW

File: widgets/toggle/__tests__/toggle-test.js [hunk 1]
```diff
@@ -3,11 +3,14 @@
 import React from 'react';
 import expect from 'expect';
 import sinon from 'sinon';
+import jsdom from 'mocha-jsdom';
 
 import toggle from '../toggle';
 import RefinementList from '../../../components/RefinementList';
 
 describe('toggle()', () => {
+  jsdom();
```
Comment (vvo) [inline widgets/toggle/__tests__/toggle-test.js]: every test that needs some DOM must call `jsdom()`, what it does is that it injects a jsdom env and remove it when the suite finishes.

**Comment by bobylito** [comment]
Feature is nice. Having a look :)

**Comment by vvo** [comment]
The new flow based on `npm run test:watch` is instant update if you modify a file, and somehow nice output (readable).

**Comment by bobylito** [comment]
LGTM :100: :dancer: :game_die: :rabbit: :tada:

**Pull request merged**